pkg/storage/iterators.go

```diff
@@ -102,38 +102,43 @@ func (u *uniqueObjectIterator) Stop() {
 }
 
 type combinedIterator[T any] struct {
-	iter1, iter2 Iterator[T]
+	iters []Iterator[T]
 }
 
 func (c *combinedIterator[T]) Next() (T, error) {
-	val, err := c.iter1.Next()
-	if err != nil {
-		if !errors.Is(err, ErrIteratorDone) {
-			return val, err
+	for i := 0; i < len(c.iters); i++ {
```

Comment:
Why not use `range`? It removes the chatty `c.iters[i]` stuff everywhere it's reference in this code.